types/react-bootstrap-table2-filter/react-bootstrap-table2-filter-tests.tsx

```diff
@@ -0,0 +1,282 @@
+    { value: 0, label: 'good' },
+    { value: 1, label: 'Bad' },
+    { value: 2, label: 'unknown' },
+];
+
+render(
+    <BootstrapTable
+        keyField="id"
+        data={products}
+        columns={[
+            { dataField: 'id', align: 'center', sort: true, text: 'Product ID' },
+            { dataField: 'name', align: 'center', sort: true, text: 'Product Name' },
+            {
+                dataField: 'quality',
+                text: 'Product Quailty',
+                // formatter: cell => selectOptionsObject[cell],
+                filter: selectFilter({
+                    options: selectOptionsObject,
+                    className: 'test-classname',
+                    withoutEmptyOption: true,
+                    defaultValue: 2,
+                    comparator: Comparator.LIKE, // default is Comparator.EQ
+                    style: { backgroundColor: 'pink' },
+                    getFilter: filter => {
+                        // qualityFilter was assigned once the component has been mounted.
+                    },
+                    onFilter: filterValue => {},
+                }),
+            },
+        ]}
+        filter={filterFactory()}
+    />,
+    document.getElementById('app'),
+);
+let qualityFilter: any;
+render(
+    <BootstrapTable
+        keyField="id"
+        data={products}
+        columns={[
+            { dataField: 'id', align: 'center', sort: true, text: 'Product ID' },
+            {
+                dataField: 'name',
+                formatter: cell => cell,
+                align: 'center',
+                sort: true,
+                text: 'Product Name',
+            },
+            {
+                dataField: 'quality',
+                text: 'Product Quailty',
+                formatter: (cell: number, row, rowIndex, formatExtraData) => selectOptionsObject[cell],
+                filter: multiSelectFilter({
+                    options: selectOptionsObject,
+                    className: 'test-classname',
+                    withoutEmptyOption: true,
+                    defaultValue: [0, 2],
+                    comparator: Comparator.LIKE, // default is Comparator.EQ
+                    style: { backgroundColor: 'pink' },
+                    getFilter: filter => {
+                        // qualityFilter was assigned once the component has been mounted.
+                        qualityFilter = filter;
+                    },
+                    onFilter: filterValue => {
+                        console.log(filterValue);
+                    },
+                }),
+            },
+        ]}
+        filter={filterFactory()}
+    />,
+    document.getElementById('app'),
+);
+
+/**
+ * Single select column test
+ */
+const selectColumns = [
+    {
+        dataField: 'quality',
+        text: 'Product Quailty',
+        formatter: (cell: number) => selectOptionsList.find(val => val.value === cell)?.value ?? '',
```

Comment:
It's the optional chaining (`?.`) in this line that's preventing it from working on versions earlier than 3.7

Reply:
Oh sorry, I don't know how I could have missed that. I just fixed that and updated the TS version to 3.0.